builtin/logical/nomad/path_config.go

```diff
@@ -0,0 +1,104 @@
+package nomad
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/vault/logical"
+	"github.com/hashicorp/vault/logical/framework"
+)
+
+func pathConfigAccess() *framework.Path {
+	return &framework.Path{
+		Pattern: "config/access",
+		Fields: map[string]*framework.FieldSchema{
+			"address": &framework.FieldSchema{
+				Type:        framework.TypeString,
+				Description: "Nomad server address",
+			},
+
+			"scheme": &framework.FieldSchema{
+				Type:        framework.TypeString,
+				Description: "URI scheme for the Nomad address",
+
+				// https would be a better default but Consul on its own
+				// defaults to HTTP access, and when HTTPS is enabled it
+				// disables HTTP, so there isn't really any harm done here.
+				Default: "http",
+			},
+
+			"token": &framework.FieldSchema{
+				Type:        framework.TypeString,
+				Description: "Token for API calls",
+			},
+		},
+
+		Callbacks: map[logical.Operation]framework.OperationFunc{
+			logical.ReadOperation:   pathConfigAccessRead,
+			logical.UpdateOperation: pathConfigAccessWrite,
+		},
+	}
+}
+
+func readConfigAccess(storage logical.Storage) (*accessConfig, error, error) {
+	entry, err := storage.Get("config/access")
+	if err != nil {
+		return nil, nil, err
+	}
+	if entry == nil {
+		return nil, fmt.Errorf(
+				"Access credentials for the backend itself haven't been configured. Please configure them at the '/config/access' endpoint"),
+			nil
+	}
+
+	conf := &accessConfig{}
+	if err := entry.DecodeJSON(conf); err != nil {
+		return nil, nil, fmt.Errorf("error reading nomad access configuration: %s", err)
+	}
+
+	return conf, nil, nil
+}
+
+func pathConfigAccessRead(
+	req *logical.Request, data *framework.FieldData) (*logical.Response, error) {
+	conf, userErr, intErr := readConfigAccess(req.Storage)
+	if intErr != nil {
+		return nil, intErr
+	}
+	if userErr != nil {
```

Comment:
If you remove this "userErr" from `readConfigAccess` just check for the nil conf below and report that it is not configured to the user.

Reply:
Just realised that no matter what happens readConfigAccess will return nil on user error, so you're right. Fixed in c4bf80c84f619df399ab08976d9ad38914c78439